library/core/src/panicking.rs

```diff
@@ -171,9 +171,9 @@ fn panic_misaligned_pointer_dereference(required: usize, found: usize) -> ! {
         super::intrinsics::abort()
     }
 
-    panic!(
+    panic_nounwind_fmt(format_args!(
```

Comment:
We should also make this function `#[rustc_nounwind]` and document that this function MUST NOT unwind.

I wouldn't worry about non-local dependency that you mentioned at https://github.com/rust-lang/rust/pull/112403#issuecomment-1591988745, since this is a lang item and it's expected that lang item may need special guarantees. A `#[rustc_nounwind]` just near the lang item attribute with a comment should be clear enough to prevent mistakes.